src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -77901,6 +77901,54 @@ static void F(bool? i, bool b)
                 Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(bool)(!i)").WithLocation(23, 13));
         }
 
+        [WorkItem(32626, "https://github.com/dotnet/roslyn/issues/32626")]
+        [Fact]
+        public void NullableCtor()
+        {
+            var source =
+@"
+#nullable enable
+
+using System;
+
+struct S
+{
+    internal object? F;
+}
+
+class Program
+{
+    static void Baseline()
+    {
+        S? x = new S();
+        x.Value.F.ToString(); // warning baseline
+
+        S? y = new S() { F = 2 };
+        y.Value.F.ToString(); // ok baseline
+    }
+
+    static void F()
+    {
+        S? x = new Nullable<S>(new S());
+        x.Value.F.ToString(); // warning
+
+        S? y = new Nullable<S>(new S() { F = 2 });
+        y.Value.F.ToString(); // ok
+    }
```

Comment:
Consider testing `new Nullable<S>()` and `new Nullable<S>(default)`.

Reply:
Consider testing `new Nullable<S>(1)`.

Reply:
`new Nullable<S>(1).`  would be an error. Is that the point of the test?

---
In reply to: [263948763](https://github.com/dotnet/roslyn/pull/33972#discussion_r263948763) [](ancestors = 263948763)

Reply:
Yes, it would be an error but it might hit the new code path with an argument type that does not match the underlying nullable type.

---
In reply to: [263951145](https://github.com/dotnet/roslyn/pull/33972#discussion_r263951145) [](ancestors = 263951145,263948763)

Reply:
Also, `new Nullable<S>(null)`.

---
In reply to: [263952567](https://github.com/dotnet/roslyn/pull/33972#discussion_r263952567) [](ancestors = 263952567,263951145,263948763)